xarray/tests/test_dataset.py

```diff
@@ -6007,6 +6005,48 @@ def test_rolling_reduce(ds, center, min_periods, window, name):
         assert src_var.dims == actual[key].dims
 
 
+@pytest.mark.parametrize("ds", (1,), indirect=True)
+@pytest.mark.parametrize("center", (True, False))
+@pytest.mark.parametrize("min_periods", (None, 1))
+@pytest.mark.parametrize("name", ("sum", "mean", "max"))
```

Comment:
Will this test will fail for something like `std`? 
IIUC iteratively applying the function to the array for each dim only works for some functions.

Which is probably fine, no need to write more tests imo, though maybe worth making a note

Reply:
In this case, many other functions should have the same result.
If there are some nans in random positions, the result should be different.
I added a few more tests and include function `std`, but I don't think it is necessary to test all nanops.